src/disk_io_thread.cpp

```diff
@@ -1633,7 +1625,6 @@ namespace libtorrent
 			j->error.operation = storage_error::read;
 			return 0;
 		}
-		j->flags |= disk_io_job::use_disk_cache;
```

Comment:
You might as well remove the block above that checks `use_read_cache` as well. Without the `use_disk_cache` flag you will hit the assert(false) in do_read in the cache disabled case.